feat(treesitter)!: incremental injection parsing (#24647)
neovim/neovim: runtime/doc/news.txt

    spaces (but paths themselves may contain spaces now).
  • |'directory'| will no longer remove a `>` at the start of the option.

> • |LanguageTree:parse()| will no longer parse injections by default and
>   now requires an explicit range argument to be passed.
> 
>   Plugins which require certain injections to be parsed must run
>   `parser:parse({ start_row, end_row })` before using the tree.

Reviewer: ```suggestion
• |LanguageTree:parse()| will no longer parse injections by default and
  now requires an explicit range argument to be passed. If injections are
  required, provide an explicit range via `parser:parse({ start_row, end_row })`.
```

• Performance:
  • 'diffopt' "linematch" scoring algorithm now favours larger and less groups
    https://github.com/neovim/neovim/pull/23611

>   • The treesitter highlighting now parses injections incrementally during
>     screen redraws for the line range being rendered. This should provide
>     significantly improved in large files with lots of injections.

Reviewer: ```suggestion
  • Treesitter highlighting now parses injections incrementally during
    screen redraws only for the line range being rendered. This significantly
    improves performance in large files with many injections.
```
